libraries/Microsoft.Bot.Builder.Dialogs/DialogSet.cs

```diff
@@ -64,6 +66,30 @@ public IBotTelemetryClient TelemetryClient
             }
         }
 
+        /// <summary>
+        /// Gets a unique string which represents the combined versions of all dialogs in this this dialogset.  
+        /// </summary>
+        /// <returns>Version will change when any of the child dialogs version changes.</returns>
+        public virtual string GetVersion()
+        {
+            if (this._version == null)
+            {
+                StringBuilder sb = new StringBuilder();
+                foreach (var dialog in this._dialogs)
+                {
+                    var v = this._dialogs[dialog.Key].GetVersion();
+                    if (v != null)
+                    {
+                        sb.Append(v);
+                    }
+                }
+
+                this._version = Convert.ToBase64String(SHA256.Create().ComputeHash(Encoding.UTF8.GetBytes(sb.ToString())));

```

Comment:
I just saw this calculation over in another file. Can we consolidate to a static method that takes a stringbuilder?

Reply:
there is no requirement for this to be standardized or documented.  Each class can use any technique it wants to create unique version id.

Reply:
I agree... We shouldn't add this to our API. If anything we could use an internal only class to ensure that its hidden from our API surface.

---
In reply to: [414933133](https://github.com/microsoft/botbuilder-dotnet/pull/3804#discussion_r414933133) [](ancestors = 414933133)